contribute/design/cstor-operator/specs.md

```diff
@@ -0,0 +1,423 @@
+      // Supported types are -- striped, mirrored, raidz and raidz2
+      poolType: striped
+    groups:
+      // Group is a collection of block devices that obeys the raid topology.
+      // For example -- cStor mirrored group can have 2 block devices only.
+
+      // Name is the name of the group.
+      // An user should put a group name and it should be unique for this
+      // node spec.
+    - name: group1
+      blockDevices:
+        // Name of the block devices can be obtained using `kubectl get bd`
+      - name: sparse-3c1fc7491f9e4cf50053730740647318
+        // id can be specified by an user and it must be unique for this node
+        // spec.
+        // If not provided by user -- it will be system generated and put.
+        id: bd1
+
+```
+
+2.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-striped
+spec:
+  nodes:
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-1
+        id: pool-2-bd1
+
+```
+
+3.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirrored
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+
+    - name: group2
+      blockDevices:
+      - name: pool-1-bd-3
+        id: pool-1-bd3
+
+      - name: pool-1-bd-4
+        id: pool-1-bd4
+
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-1
+        id: pool-2-bd1
+
+      - name: pool-2-bd-2
+        id: pool-2-bd2
+
+```
+4.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-hybrid
+spec:
+  nodes:
+
+  - name: gke-cstor-it-default-pool-1
+    poolSpec:
+      poolType: mirrored
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-1-bd-1
+        id: pool-1-bd1
+
+      - name: pool-1-bd-2
+        id: pool-1-bd2
+
+    - name: group2
+      blockDevices:
+      - name: pool-1-bd-3
+        id: pool-1-bd3
+
+      - name: pool-1-bd-4
+        id: pool-1-bd4
+
+  - name: gke-cstor-it-default-pool-2
+    poolSpec:
+      poolType: striped
+
+    groups:
+    - name: group1
+      blockDevices:
+      - name: pool-2-bd-2
+        id: pool-2-bd2
+
+```
+
+Following is a sample auto CStorPoolCluster specs
+
+1.
+
+```
+
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-sparse-pool-test
+spec:
+  maxPools: 3
```

Comment:
can this be `maxCount`

Reply:
cc @vishnuitta -- Can we discuss

Reply:
lets close this when we three discussing with @kmova

Reply:
Let us not bring in the auto spec yet. Once we finish the Manual cStorPoolCluster - we can check on the things that really need to be available. Also, to avoid confusion - may be we should not have the same spec used for manual and auto.

Reply:
Done